### ISSUE TITLE
New contents do not render after emptying scene

### ISSUE THREAD

**Issue by dominic-chang** [MakieOrg/Makie.jl#4086]
- [Yes] are you running newest version (version from docs) ?
- [Yes] can you reproduce the bug with a fresh environment ? (`]activate --temp; add Makie`)
- [julia 1.10.4 Mac M3 ] What platform + GPU are you on?

New contents that are added to a scene after calling `empty!` do not render in `CairoMakie` or `GLMakie`.

 Here is a minimum working example:

```
using CairoMakie

fig = Figure()
ax = Axis(fig[1,1])
scatter!(ax, 1,1)
display(fig)
```
<img width="493" alt="image" src="https://github.com/user-attachments/assets/eaeeb6f7-9c6c-4b6c-b228-367207baf3b2">

```
empty!(ax.scene)
scatter!(ax, 1,1)
display(fig)
```
<img width="493" alt="image" src="https://github.com/user-attachments/assets/49e26767-0001-4191-bfe4-7f06063121ed">

I did not have this issue with `WGLMakie`

**Comment by asinghvi17** [comment]
This is because `empty!` also removes the Scene from the scene graph.
```julia
julia> ax.blockscene.children
Scene[]
```

Arguably this is then some strange behaviour in WGLMakie where the scene is not dropped from the graph.  But empty should probably not remove the Scene entirely either.

For now, you can `empty!(ax.scene.plots)` for the "correct" effect:

```julia
using CairoMakie

fig = Figure()
ax = Axis(fig[1,1])
scatter!(ax, 1,1; color = :red)
display(fig)
empty!(ax.scene.plots)
scatter!(ax, 1,1; color = :blue)
display(fig)
```

**Comment by SimonDanisch** [comment]
@asinghvi17 I dont think so, `empty!(ax.scene.plots)` doesn't remove the plots from being displayed (in GLMakie) and it does not delete the scene from the scene graph:
![image](https://github.com/user-attachments/assets/f48ce464-fc49-43ae-b20a-20041857d2dc)

Also we do have tests for this, and I actually can't reproduce this on GLMakie@0.10.5.

Maybe this is an M3 issue? Also, you should be using `empty!(ax)`, can you try if that works?

**Comment by SimonDanisch** [comment]
For CairoMakie, empty!(ax.scene) indeed does not work which is a bit weird, but you should really be using `empty!(ax)`

**Comment by ffreyer** [comment]
https://github.com/MakieOrg/Makie.jl/blob/6a52067c924b3ea28b160dd5a92b3eda961e477c/src/scenes.jl#L429-L431
This is removing *this* scene from its parent without triggering any backend cleanup.

**Comment by dominic-chang** [comment]
> For CairoMakie, empty!(ax.scene) indeed does not work which is a bit weird, but you should really be using `empty!(ax)`

Thanks. This produced the behaviour I was expecting.

**Comment by dominic-chang** [comment]
> https://github.com/MakieOrg/Makie.jl/blob/6a52067c924b3ea28b160dd5a92b3eda961e477c/src/scenes.jl#L429-L431
> 
> 
> This is removing _this_ scene from its parent without triggering any backend cleanup. So if the backend hasn't displayed the scene yet it won't see it from the scene tree. If it was already displayed it'll likely continue being displayed.

Is this not a bug then? Should I close this issue?
